### DISCUSSION TITLE
Interpolação, locals e output

### DISCUSSION THREAD

**Discussion by nelsonstudent** [chgasparoto/curso-aws-com-terraform#70] [Q&A]
Olá!

Ao rodar o comando terraform plan -out="tfplan.out" surge a seguinte mensagem:

**_Error: Reference to undeclared resource 
       on s3.tf line 2, in resource "aws_s3_bucket" "this":
        2:      bucket  =  "${random_pet.bucket.id}-${var.environment}"
A managed resource "random_pet" "bucket" has not beeb declared in the root module._**

Que módulo root é esse onde eu não declarei o random_pet ?

**Comment by chgasparoto** [comment]
@nelsonstudent o root module é onde está todos os arquivos do Terraform e o diretório que você está executando os comandos. No caso, esta pasta https://github.com/chgasparoto/curso-aws-com-terraform/tree/master/01-terraform-basico/04-interpolacao-locals-outputs

Sobre o erro acima, me parece que está faltando este recurso para você:

https://github.com/chgasparoto/curso-aws-com-terraform/blob/master/01-terraform-basico/04-interpolacao-locals-outputs/random.tf

**Comment by nelsonstudent** [comment]
@chgasparoto o recurso que você disse que achava que está faltando, está presente, no entanto, a relação de arquivos que você mostra no link https://github.com/chgasparoto/curso-aws-com-terraform/tree/master/01-terraform-basico/04-interpolacao-locals-outputs eu não os vi durante o curso (até a presente aula). Como por exemplo o arquivo locals.tf. 

Vou rever esta aula para ver se você informou e eu não o vi.

**Comment by nelsonstudent** [comment]
@chgasparoto descobri o erro. O arquivo random_pet estava fora do bloco, por isso o Terraform não o encontrava. 

Tudo resolvido. 

Obrigado pela ajuda.

**Reply by chgasparoto** [reply]
Top @nelsonstudent , abraço.
